src/cheri_pte.sail

```diff
@@ -105,8 +105,39 @@ function checkPTEPermission_LC(p, e, ext_ptw) =
       then PTE_Check_Success(ext_ptw_lc_join(ext_ptw, PTW_LC_OK))
       else PTE_Check_Success(ext_ptw_lc_join(ext_ptw, PTW_LC_TRAP))
     }
+  };
+
+  match (res) {
+    PTE_Check_Failure(_, _) => res,
+    PTE_Check_Success(ext_ptw) =>
+      if ext_ptw.ptw_lc == PTW_LC_TRAP then {
+        if scisa.pte_cap_lc_datadep() == 0b1 then {
+          /*
+           * Continue translating, but add caveat
+           */
+          PTE_Check_Success(ext_ptw)
+        } else {
+          /*
+           * Stop translating and raise a trap now
+           */
+          PTE_Check_Failure(ext_ptw, EPTWF_CAP_ERR)
+        }
+      } else res
   }
+}
 
+val checkPTEPermission_SC : (Ext_PTE_Bits, ext_ptw) -> PTE_Check effect { rreg }
+function checkPTEPermission_SC(e, ext_ptw) = {
+  if   e.CapWrite() == 0b1
+  then PTE_Check_Success(ext_ptw)
+  else {
+         let ext_ptw' : ext_ptw = ext_ptw_sc_join(ext_ptw, PTW_SC_TRAP);
+
+         if   scisa.pte_cap_sc_datadep() == 0b1
```

Comment:
Should this line check the tag?

Reply:
Oh, sorry, I see that we return a value indicating that a tag should imply a trap, but I don't see where the check of that tag actually happens.

Reply:
Indeed, I think this doesn't work as I placed it here.  The data dependence is supposed to be brought forward into translation by the access type (`Write(Data)` vs `Write(Cap)`) and is ultimately analysed by `translationException` and so the right place for `pte_cap_sc_datadep` is in `cheri_instr`.  One revised commit coming right up.